extensions/amp-inline-gallery/0.1/amp-inline-gallery-thumbnails.js

```diff
@@ -0,0 +1,241 @@
+    const content = html`
+      <div class="i-amphtml-inline-gallery-thumbnails-container">
+        <div class="i-amphtml-inline-gallery-thumbnails-thumbnail">
+          <div class="i-amphtml-inline-gallery-thumbnails-resizer"></div>
+        </div>
+      </div>
+    `;
+
+    // If an thumb width/height aspect ratio specified, used that. Otherwise
+    // use the aspect ratio of the element we are creating a thumb and
+    // fall back to a square if anything goes wrong.
+    const srcAspectRatio =
+      srcElement./*OK*/ offsetWidth / srcElement./*OK*/ offsetHeight;
+    const aspectRatio = this.thumbAspectRatio_ || srcAspectRatio || 1;
+    // Use a padding-right (along with `writing-mode: vertical-lr` in the CSS)
+    // to make the width match the aspect ratio for the available height. Note
+    // that we do not use an `svg` with a `viewBox` as that has problems in
+    // Firefox.
+    setStyle(
+      content.querySelector('.i-amphtml-inline-gallery-thumbnails-resizer'),
+      'padding-right',
+      100 * aspectRatio,
+      '%'
+    );
+
+    content
+      .querySelector('.i-amphtml-inline-gallery-thumbnails-thumbnail')
+      .appendChild(this.getThumbnailContent_(srcElement));
+    content.onclick = () => {
+      this.element.dispatchEvent(
+        createCustomEvent(
+          this.win,
+          InlineGalleryEvents.GO_TO_SLIDE,
+          dict({
+            'index': index,
+          }),
+          {
+            bubbles: true,
+          }
+        )
+      );
+      this.carousel_.getImpl().then(impl => {
+        impl.goToSlide(index, {smoothScroll: true});
+      });
+    };
+    return content;
+  }
+
+  /**
+   * @return {!Element} An element to use when there is no image available
+   *    for the thumbnail.
+   */
+  createDefaultThumbnail_() {
+    const div = document.createElement('div');
+    div.className = 'i-amphtml-inline-gallery-thumbnails-default';
+    return div;
+  }
+
+  /**
+   * Creates a thumbnail element for a given slide. If the slide is an image
+   * directly contains an image, then the image is used. Otherwise, a
+   * placeholder element is created instead.
+   * @param {!Element} slide
+   * @return {!Element}
+   */
+  getThumbnailContent_(slide) {
+    const image = matches(slide, 'amp-img, img')
+      ? slide
+      : scopedQuerySelector(slide, '> amp-img, > img');
+
+    if (!image) {
+      return this.createDefaultThumbnail_();
+    }
+
+    const thumbImg = document.createElement('amp-img');
+    thumbImg.className = 'i-amphtml-inline-gallery-thumbnails-image';
+    thumbImg.setAttribute('layout', 'fill');
+
+    // Use the attribute since this may be an amp-img.
+    const src = image.getAttribute('src');
+    if (src) {
+      thumbImg.setAttribute('src', src);
+    }
+
+    // Use the attribute since this may be an amp-img.
+    const srcset = image.getAttribute('srcset');
+    if (srcset) {
+      thumbImg.setAttribute('srcset', srcset);
+    }
```

Comment:
Why not clone?

Reply:
Added a comment above. Don't want to copy over any styling (via classes, styles or attributes).